core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/ScanSourceOpDesc.scala

```diff
@@ -87,16 +83,20 @@ abstract class ScanSourceOpDesc extends SourceOperatorDescriptor {
 
   def inferSchema(): Schema
 
-  // resolve the file path based on whether the user system is enabled
-  // it will check for the presence of the given filePath/Desc
+  // get the source file descriptor from the fields
+  // either the datasetFile or the filePath should be defined
   def determineFilePathOrDatasetFile(): (String, DatasetFileDocument) = {
-    if (AmberConfig.isUserSystemEnabled) {
-      val file = datasetFile.getOrElse(
-        throw new RuntimeException("Dataset file descriptor is not provided.")
-      )
+    if (
```

Comment:
Use Either can relax this both None / both Some check

Reply:
changed